src/pleiades/utils/timepix.py

```diff
@@ -0,0 +1,258 @@
+
+    if facility == Facility.ornl:
+        update_with_spectra_files_at_ornl(master_dict, normalization_status)
+    elif facility == Facility.lanl:
+        update_with_spectra_files_at_lanl(master_dict, normalization_status)
+    else:
+        # Implement the logic for other facilities if needed
+        pass
+
+
+def update_with_spectra_files_at_ornl(master_dict: dict, normalization_status: NormalizationStatus) -> None:
+    """
+    Update the master dictionary with the spectra files for ORNL.
+    """
+    for data_path in master_dict[MasterDictKeys.list_folders].keys():
+        spectra_files = glob.glob(os.path.join(data_path, "*_Spectra.txt"))
+        if len(spectra_files) == 0:
+            logger.info(f"\tNo spectra file found for {data_path}")
+            return
+
+        else:
+            spectra_file = spectra_files[0]
+            pd_spectra = pd.read_csv(spectra_file, sep=",", header=0)
+            shutter_time = pd_spectra["shutter_time"].values
+            master_dict[MasterDictKeys.list_folders][data_path][MasterDictKeys.list_spectra] = shutter_time
+
+    normalization_status.all_spectra_file_found = True
+
+
+def update_with_spectra_files_at_lanl(master_dict: dict, normalization_status: NormalizationStatus) -> None:
+    """
+    Update the master dictionary with the spectra files for LANL.
+    for LANL, the time spectra can be build up using the metadata contained in the files name.
+
+    for example:
+    file_name = image_m2M9997Ex512y512t1e6T2000p1e6P100.tiff
+    where T2000 is the number of time bins
+          t1e6 is the time bin size in microseconds (1e6 -> 1e-6)
+    """
+    for folder in master_dict[MasterDictKeys.list_folders].keys():
+        list_files = master_dict[MasterDictKeys.list_folders][folder][MasterDictKeys.list_images]
+
+        # extract number of time bins and time bin size from the first file name
+        if len(list_files) == 0:
+            logger.info(f"\tNo spectra file found for {folder}")
+            return
+
+        first_file_name = os.path.basename(list_files[0])
+        number_of_frames = retrieve_number_of_frames_from_file_name(first_file_name)
+        bin_size = retrieve_time_bin_size_from_file_name(first_file_name)
+
+        # create the time spectra
+        time_spectra = np.arange(0, number_of_frames * bin_size, bin_size)
+        master_dict[MasterDictKeys.list_folders][folder][MasterDictKeys.list_spectra] = time_spectra
+
+
+def update_with_shutter_values(
+    master_dict: dict, normalization_status: NormalizationStatus, facility=Facility.ornl
+) -> None:
+    """
+    Update the master dictionary with the shutter values.
+    """
+    logger.info(f"Updating {master_dict[MasterDictKeys.data_type]} master dictionary with shutter values")
+    if facility == Facility.ornl:
+        update_with_shutter_values_at_ornl(master_dict, normalization_status)
+    else:
+        # Implement the logic for other facilities if needed
+        pass
```

Comment:
I think we should contact Alex to see if LANL is applying similar correction, and maybe throw a warning here is the facility is not ORNL, and indicating that shutter value correction is skipped for non-ORNL dataset.